### PR TITLE
RavenDB-7589

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/RestoreBackupTask.cs [hunk 1]
```diff
@@ -139,7 +140,7 @@ public class RestoreBackupTask
                     if (snapshotRestore)
                         options |= InitializeOptions.GenerateNewDatabaseId;
                     database.Initialize(options);
-                    SmugglerRestore(_restoreConfiguration.BackupLocation, database, databaseRecord, onProgress, restoreResult);
+                    SmugglerRestore(_restoreConfiguration.BackupLocation, database, databaseRecord, restoreSettings.Identities, onProgress, restoreResult);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/PeriodicBackup/RestoreBackupTask.cs]: Need to save identities here. Will do that tomorrow.

File: src/Raven.Server/Documents/PeriodicBackup/RestoreBackupTask.cs [hunk 2]
```diff
@@ -271,6 +272,7 @@ private List<string> GetFilesForRestore(string backupLocation)
             string backupDirectory, 
             DocumentDatabase database, 
             DatabaseRecord databaseRecord,
+            Dictionary<string, long> identities,
```
Comment (ayende) [inline src/Raven.Server/Documents/PeriodicBackup/RestoreBackupTask.cs]: What if we have a _lot_ of them?

File: src/Raven.Server/Documents/Handlers/Debugging/IdentityDebugHandler.cs [hunk 3]
```diff
@@ -14,15 +13,10 @@ public Task GetIdentities()
...
             {
-                var record = Database.ServerStore.Cluster.ReadDatabase(context, Database.Name);
-                var identitiesAsJson = record.Identities.ToJson();
-
-                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
-                {
-                    context.Write(writer, identitiesAsJson);
-                    writer.Flush();
-                }
+                var json = Database.ServerStore.Cluster.Read(context, Constants.Documents.IdentitiesPrefix + Database.Name.ToLowerInvariant());
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Debugging/IdentityDebugHandler.cs]: This assume a single object holding all identities per database, right?
That is going to be a problem if we'll have users with thousands of identities. Better to use a value per identity per db.
So the key is going to be `db/db-name/identities/identity-name`.
The idea is that deleting the db will also delete the identities.

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by ppekrol** [comment]
Changes applied.

**Pull request merged**